app/services/edge_cache/buster.rb

```diff
@@ -0,0 +1,115 @@
+module EdgeCache
+  class Buster
+    PROVIDERS = %w[fastly nginx].freeze
+
+    def initialize(path)
+      # TODO: (Vaidehi Joshi) - Right now, we are checking that nginx is
+      # available on every purge request/call to this buster service. If we are going
+      # to bust multiple paths, we should be able to check that nginx is
+      # available just once, and persist it on the class with @provider_available?.
+      # Then, we could allow for an array of @paths = [] to be passed in,
+      # and on single buster instance could bust multiple paths in order.
+
+      @path = path
+      @provider = set_provider
+    end
+
+    def self.call(*args)
```

Comment:
@citizen428 To be honest with you, I also found it kind of unnecessarily confusing, too. The tests took me _much_ longer than they should have because I had a hard time keeping track of whether I was dealing with a class `call` or an instance `call` 🙊 

I just followed the convention of the other service classes since I figured that deviating from that would bring up a lot of comments during code review (especially because this is adding a new file)

Reply:
Tbh, the only comments this seems to bring up is me saying we stop doing it every time I see it 😉